packages/client-utils/src/sync-tools.js

```diff
@@ -48,19 +49,50 @@ export const retryUntilCondition = async (
   operation,
   condition,
   message,
-  { maxRetries = 6, retryIntervalMs = 3500, log = console.log, setTimeout },
+  {
+    maxRetries = 6,
+    retryIntervalMs = 3500,
+    reusePromise,
+    log = console.log,
+    setTimeout,
```

Comment:
out of scope but I think we should be careful to avoid configuration parameters from ocaps.

consider,
```suggestion
    // XXX mixes ocaps with configuration options
    log = console.log,
    setTimeout,
```